model/build-processor/src/main/java/org/keycloak/models/map/processor/GenerateEntityImplementationsProcessor.java

```diff
@@ -283,28 +284,26 @@ public void generate(TypeElement e) throws IOException {
                   .map(ExecutableElement.class::cast)
                   .filter((ExecutableElement ee) -> ee.getKind() == ElementKind.CONSTRUCTOR)
                   .forEach((ExecutableElement ee) -> {
-                      if (hasDeepClone || ! needsDeepClone) {
-                          pw.println("    "
-                            + ee.getModifiers().stream().map(Object::toString).collect(Collectors.joining(" "))
-                            + " " + mapSimpleClassName + "(" + methodParameters(ee.getParameters()) + ") { super(" + ee.getParameters() + "); }"
-                          );
-                      } else if (needsDeepClone) {
+                      // Create constructor and initialize cloner to DUMB_CLONER if necessary
+                      if (usingGeneratedCloner) {
                           pw.println("    /**");
                           pw.println("     * @deprecated This constructor uses a {@link DeepCloner#DUMB_CLONER} that does not clone anything. Use {@link #" + mapSimpleClassName + "(DeepCloner)} variant instead");
                           pw.println("     */");
-                          pw.println("    "
-                            + ee.getModifiers().stream().map(Object::toString).collect(Collectors.joining(" "))
-                            + " "
-                            + mapSimpleClassName + "(" + methodParameters(ee.getParameters()) + ") { this(DeepCloner.DUMB_CLONER" + (ee.getParameters().isEmpty() ? "" : ", ") + ee.getParameters() + "); }"
-                          );
-                          pw.println("    "
-                            + ee.getModifiers().stream().map(Object::toString).collect(Collectors.joining(" "))
-                            + " "
-                            + mapSimpleClassName + "(DeepCloner cloner" + (ee.getParameters().isEmpty() ? "" : ", ") + methodParameters(ee.getParameters()) + ") { super(" + ee.getParameters() + "); this.cloner = cloner; }"
-                          );
                       }
+                      pw.println("    "
+                              + ee.getModifiers().stream().map(Object::toString).collect(Collectors.joining(" "))
+                              + " " + mapSimpleClassName + "(" + methodParameters(ee.getParameters()) + ") {"
+                      );
+                      pw.println("        super(" + ee.getParameters() + ");");
+                      if (usingGeneratedCloner) pw.println("        this.cloner = DeepCloner.DUMB_CLONER;");
+                      pw.println("    }");
                   });
 
+                pw.println("    "
+                        + "public "
+                        + mapSimpleClassName + "(DeepCloner cloner) { super(); " + (!usingGeneratedCloner ? "" : "this.cloner = cloner;") + "}"
```

Comment:
What if `super()` does not exist?

Reply:
The compilation would fail. But it sounds like correct behavior to me as all these classes need to be instantiable by `DeepCloner` that is not able to provide any parameters.